engine/runtime/src/main/java/org/enso/interpreter/runtime/type/Types.java

```diff
@@ -113,23 +113,29 @@ public static void extractArguments(Object[] arguments) throws ArityException {
    */
   public static String getName(Object value) {
     if (TypesGen.isLong(value) || TypesGen.isImplicitLong(value)) {
-      return "Number";
+      return Constants.INTEGER;
+    } else if (TypesGen.isDouble(value)) {
+      return Constants.DECIMAL;
+    } else if (TypesGen.isEnsoBigInteger(value)) {
+      return Constants.NUMBER;
```

Comment:
No, this case is also `Constants.Integer`. The numerical type hierarchy is `Decimal <: Number`, `Integer <: Number`, `Integer = long | BigInteger`. `Number` doesn't exist at runtime at all. It's just a convenient "superclass" for the basic numeric types.